core/corehttp/gateway_handler_car.go

```diff
@@ -67,6 +67,7 @@ func (i *gatewayHandler) serveCAR(ctx context.Context, w http.ResponseWriter, r
 
 	w.Header().Set("Content-Type", "application/vnd.ipld.car; version=1")
 	w.Header().Set("X-Content-Type-Options", "nosniff") // no funny business in the browsers :^)
+	w.Header().Set("Trailer", "X-Stream-Error")
```

Comment:
I belive it's missing on other types of responses too (directories and files)

Reply:
CAR is the only handler that sets the `X-Stream-Error` header in case of error. The others don't, so they wouldn't need the `Trailer` Header.

Reply:
@hacdias unixfs files are also streaming, so we should them to also emit thoses then.

Reply:
@Jorropo indeed. The main difference here is that the UnixFS streaming can leverage the standard `http.ServeContent` function, which handles some of the errors by itself. Ultimately, when it actually outputs the file via `io.Copy`, it also ignores the errors. Since the function doesn't return the error, there is not much we can do.

However, I would argue that the possible errors between both are different since the errors from `io.Copy` can be something unexpected like lost connections. Here it's an error that we know can happen but are only able to detect mid-stream and we want to avoid the user to think they received the entire file. I'll also add you on the PR reviews to get your feedback.